src/bench/inputfetcher.cpp

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2024-present The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <bench/bench.h>
+#include <bench/data/block413567.raw.h>
+#include <coins.h>
+#include <common/system.h>
+#include <inputfetcher.h>
+#include <primitives/block.h>
+#include <serialize.h>
+#include <streams.h>
+#include <util/time.h>
+
+static constexpr auto QUEUE_BATCH_SIZE{128};
+static constexpr auto DELAY{2ms};
+
+//! Simulates a DB by adding a delay when calling GetCoin
+class DelayedCoinsView : public CCoinsView
```

Comment:
In 2349ac7d6071746a80223358bce0d5e556b277d7 "bench: add inputfetcher bench"
nit: will be nice if we have `block413567` input's data that we can read so that we dont have to simulate this.

Reply:
We're reading the previous outpoints of that block's inputs, which are in many other previous blocks. So, not sure this is feasible.